static/js/compose_error.ts

```diff
@@ -1,11 +1,68 @@
 import $ from "jquery";
 
+import render_compose_banner from "../templates/compose_banner.hbs";
+import render_stream_does_not_exist_error from "../templates/stream_does_not_exist_error.hbs";
+
 import * as common from "./common";
 
 // banner types
 export const WARNING = "warning";
+export const ERROR = "error";
+
+const WARNING_MESSAGES = [".topic_resolved"];
+
+const ERROR_MESSAGES = [
+    ".empty_message",
+    ".wildcards_not_allowed",
+    ".subscription_error",
+    ".stream_does_not_exist",
+    ".missing_stream",
+    ".no_post_permissions",
+    ".private_messages_disabled",
+    ".missing_private_message_recipient",
+    ".invalid_recipient",
+    ".invalid_recipients",
+    ".deactivated_user",
+    ".message_too_long",
+    ".topic_missing",
+    ".zephyr_not_running",
+    ".generic_compose_error",
+];
+
+export const ALL_BANNER_MESSAGES = WARNING_MESSAGES.concat(ERROR_MESSAGES);
+
+// TODO: Replace with compose_ui.hide_compose_spinner() when it is converted to ts.
+function hide_compose_spinner(): void {
+    $("#compose-send-button .loader").hide();
+    $("#compose-send-button span").show();
+    $("#compose-send-button").removeClass("disable-btn");
+}
+
+// NOTE: if you add a new classname, make sure it's added to ALL_BANNER_MESSAGES above.
```

Comment:
What happens if it isn't? I think it's reasonable to have that case throw a `blueslip.error`.

The other direction we could consider is having `ERROR_MESSAGES` be a map, and the caller is supposed to pass, for example, `compose_banner.ERROR_MESSAGES.invalid_recipient`; then it'd be hard to write code that didn't pass a value in that object.